components/admin/contentPage/Footer.tsx

```diff
@@ -18,6 +37,73 @@ const VisuallyHiddenInput = styled("input")({
   width: 1,
 });
 const Footer: React.FC = () => {
+  const [formData, setFormData] = useState<FooterFormData>({
+    logo: null,
+    description: "",
+    ownerOfCopyright: "",
+    footerSection1: "",
+    footerSection1SubSection1: "",
+    footerSection1SubSection2: "",
+    footerSection1SubSection3: "",
+    footerSection2: "",
+    footerSection2SubSection1: "",
+    footerSection2SubSection2: "",
+    footerSection2SubSection3: "",
+    footerSection3: "",
+    footerSection3SubSection1: "",
+    footerSection3SubSection2: "",
+    footerSection3SubSection3: "",
+  });
+  const [errors, setErrors] = useState<Partial<Record<keyof FooterFormData, string>>>({}); // Define the state type as a partial record and Start with an empty object to hold errors for each form field
+
+    // Ensure file-related code only runs in the browser
+    useEffect(() => {
+      if (typeof window === "undefined") {
+        console.error("Running in a non-browser environment.");
+      } else if (typeof File === "undefined") {
+        console.error("File API is not supported in this browser.");
+      }
+    }, []);
+    
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target; // Extract the name and value from the input event
+    setFormData((prevData) => ({ ...prevData, [name]: value })); // Update the form data state with the new value
+  };
+
+  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const files = e.target.files; // Extract the files from the input event
+    // Update the form data state with the selected files
+    setFormData((prevData) => ({
+      ...prevData, // Spread the previous state to retain other fields
+      logo: files ? Array.from(files) : [], // Handle files or default to an empty array
+    }));
+  };
+
+  const handleSubmit = () => {
+    // Ensure formData is properly initialized
+    const adjustedData = {
+      ...formData,
+      logo: formData.logo || [], // Convert null to empty array
+    };
+
+    const validation = schemaAdminPanelContentFooter.safeParse(adjustedData);
```

Comment:
Why have you not used the common validation function? Please reuse the validation function in utils/validation.ts